### ISSUE TITLE
Add additional "sigmoid" approximation to GeLu activation?

### ISSUE THREAD

**Issue by kklocker** [pytorch/pytorch#102447]
### 🚀 The feature, motivation and pitch

Hi,

I was just ranomly browsing [tinygrad](https://github.com/geohot/tinygrad) and saw the "quick_gelu" function, noticing also that this implementation is not a part of pytorch it self.

Pytorch only seems to have the `tanh` - approximation, so this is a feature request to add the second approximation also mentioned in the paper [Gaussian Error Linear Units (GELUs)](https://arxiv.org/abs/1606.08415v4)

I am not a AI expert, but this was just a simple thing I noticed and could gladly implement :)

### Alternatives

* Add a `sigmoid` approximation also, `x*sigmoid(1.702 * x)`
* Do nothing :)

### Additional context

Basically this:

https://github.com/kklocker/pytorch/tree/feature/add-gelu-approximation

**Comment by jbschlosser** [comment]
Hey @kklocker, thanks for the request! While such an approximation seems potentially valuable, every addition to PyTorch Core comes with a significant implementation and maintenance cost. In general, we take a fairly conservative stance towards new additions for this reason.

FWIW, I was the internal dev responsible for introducing the tanh approximation, and it was a large headache to introduce due to BC issues. If there hadn't been a perf benefit to introducing it, it likely would not have been worth it.

I think it's worth leaving this issue open for now to gather community feedback around whether an addition like this would be valuable and worthwhile to implement and maintain. If we find clear benefits to a PyTorch Core-maintained implementation of this approximation, we'd certainly to be willing to consider it.

**Comment by kklocker** [comment]
@jbschlosser Hi, sorry for not responding earlier.

I completely understand your concerns, and I think the conservative stance is a rather healthy sign of pytorch

My assumptions would be that basically all the necessary ground work would have been done in the introduction of the tanh-approx, and that adding another optional (non-default) parameter now would be rather straight forward and "cheap" -- but of course this might be a fairly naive assumption, given a rather lacking insight to the code base.

Out of curiosity, do you know what the cause(s) of the "large headache" caused by the tanh-approx was? (as hinted at above this is not something that is very important to me, just curious 😊)

^ Also noticing that others have noticed this thing is missing from pytorch
